packages/rum-core/src/domain/resource/resourceCollection.ts

```diff
@@ -232,3 +234,12 @@ function computeRequestDuration(pageStateHistory: PageStateHistory, startClocks:
 function discardZeroStatus(statusCode: number | undefined): number | undefined {
   return statusCode === 0 ? undefined : statusCode
 }
+
+/**
+ * The 'nextHopProtocol' is an empty string for cross-origin resources without CORS headers,
+ * meaning the protocol is unknown, and we shouldn't report it.
+ * https://developer.mozilla.org/en-US/docs/Web/API/PerformanceResourceTiming/nextHopProtocol#cross-origin_resources
+ */
+function discardEmptyProtocol(requestProtocol: string | undefined): string | undefined {
+  return requestProtocol === '' ? undefined : requestProtocol
+}
```

Comment:
**💬 suggestion:** to refactor things a bit further, ‏move this function to `resourceUtils.ts` and change it like this:

```ts
export function computeResourceEntryProtocol(entry: RumPerformanceResourceTiming) {
  return entry.nextHopProtocol === '' ? undefined : entry.nextHopProtocol
}
```